presto-main/src/main/java/io/prestosql/operator/ScanFilterAndProjectOperator.java

```diff
@@ -242,16 +245,23 @@ else if (cursor != null) {
 
             checkState(cursor == null && pageSource == null, "Table scan split already set");
 
-            if (!dynamicFilter.get().isAll()) {
-                dynamicFilterSplitsProcessed++;
+            ListenableFuture<?> dynamicFilterFuture = dynamicFilterSupplier.getBlockingFuture();
+            if (!dynamicFilterFuture.isDone()) {
+                return blocked(dynamicFilterFuture);
```

Comment:
Something to watch out for is who adds listeners to this future and how many times this happens. We've seen problems in the past where we "leak" registrations (rather, add too many listeners, which end up causing performance problems and OOM)

Reply:
IIUC, we propagate this future to `Driver#firstFinishedFuture`, which registers a single listener for each blocked operator future:
https://github.com/prestosql/presto/blob/6541b8fd0bcb05b86755274b495673d67b4dea37/presto-main/src/main/java/io/prestosql/operator/Driver.java#L635-L648
(found by running `testJoinDynamicFilteringBlockProbeSide` integration test)

Not sure though if it's the only place this future is being used.